pallets/msa/src/lib.rs

```diff
@@ -404,6 +406,17 @@ impl<T: Config> Pallet<T> {
 		Ok(())
 	}
 
+	pub fn ensure_valid_delegation(provider: Provider, delegator: Delegator) -> DispatchResult {
+		let current_block = frame_system::Pallet::<T>::block_number();
+		let info = Self::get_provider_info_of(provider, delegator)
+			.ok_or(Error::<T>::DelegationNotFound)?;
+		if info.expired == T::BlockNumber::zero() {
+			return Ok(())
+		}
+		ensure!(info.expired >= current_block, Error::<T>::DelegationExpired);
```

Comment:
I think it probably should be `info.expired > current_block`. Maybe shorten it like the following

`ensure!(info.expired == T::BlockNumber::default() || info.expired > current_block, Error::<T>::DelegationExpired);`